src/bin/jp2/opj_compress.c

```diff
@@ -485,6 +486,9 @@ static unsigned int get_num_images(char *imgdirpath)
         if (strcmp(".", content->d_name) == 0 || strcmp("..", content->d_name) == 0) {
             continue;
         }
+        if (num_images == UINT_MAX) {
+            fprintf(stderr, "Too many files in folder %s\n", imgdirpath);
```

Comment:
missing break;

Reply:
Added return statement

Reply:
if you use return, you need an explicit closedir(dir) before.
Other possibilities do:
num_images = 0;
break;
Same for other instances

Reply:
Ah got you, for whatever reason I was thinking the goto fin statement in the calling section of code cleaned everything up, including the dir.  Also didn't realize the opendir function closed the directory if there was nothing there and freed the memory. I made the correction as requested. Thanks again.